database/database_test.go

```diff
@@ -239,6 +275,49 @@ func TestStatesWithStatus(t *testing.T) {
 	}
 }
 
+func TestSetAutomatedPendingRemoval(t *testing.T) {
+	resetDB()
+
+	testStates := []DomainState{
+		{Name: "preloaded.test", Status: StatusPreloaded},
+		{Name: "pending.test", Status: StatusPending},
+		{Name: "pendingremoval.test", Status: StatusPendingRemoval},
+		{Name: "rejected.test", Status: StatusRejected},
+		{Name: "removed.test", Status: StatusRemoved},
+		{Name: "unknown.test", Status: StatusUnknown},
+		{Name: "pendingautomatedremoval.test", Status: StatusPendingAutomatedRemoval}}
+
+	for _, state := range testStates {
+		if err := testDB.PutState(state); err != nil {
+			t.Errorf("cannot put state %s", err)
+			return
+		}
+	}
+
+	testDomains := []string{"preloaded.test", "pending.test", "pendingremoval.test", "rejected.test", "removed.test", "unknown.test", "pendingautomatedremoval.test"}
+	var statuses []string
+	statusReport := func(format string, args ...interface{}) {
+		formatted := fmt.Sprintf(format, args...)
+		statuses = append(statuses, formatted)
+	}
+
+	err := testDB.SetPendingAutomatedRemoval(testDomains, statusReport)
+	if err != nil {
+		t.Errorf("Error for test SetAutoamtedPendingRemoval: %s", err)
```

Comment:
nit: typo in SetAutomatedPendingRemoval (also consider making this a bit more descriptive (e.g. "Unexpected error in SetAutomatedPendingRemoval").

Reply:
Done